### DISCUSSION TITLE
How to compressed file in Minio (Spring, Docker)

### DISCUSSION THREAD

**Discussion by Limbo1111** [minio/minio#12729] [Q&A] [answered]
I have a Spring project that runs in Docker, back, web are launched through the docker-compose file, minio is also there, I don't know how to set up compression, I tried all the different options, it doesn't work. Please help)

This is my Docker-compose file

environment:
      MINIO_ACCESS_KEY: minio
      MINIO_SECRET_KEY: miniominio
      MINIO_COMPRESS: on
      MINIO_COMPRESS_EXTENSIONS:
      MINIO_COMPRESS_MIME_TYPES:
or it might be worth placing compression in the application.properties.yml

**Comment by klauspost** [comment, marked as answer]
@Limbo1111 Use `mc admin config set MYMINIO compression enable=on extensions= mime_types=`.

Note, by default compression is disabled on encrypted objects. To enabled that, add `allow_encryption=on` as well.
